src/host-session-service.vala

```diff
@@ -345,6 +345,29 @@ namespace Frida {
 			throw new Error.INVALID_OPERATION ("Only meant to be implemented by services");
 		}
 
+		public async HashTable<string, Variant>? query_system_parameters (Cancellable? cancellable) throws Error, IOError {
+			#if DARWIN && !IOS
+				Variant? platform = new Variant ("s", "macos");
```

Comment:
Would be good to declare `string os;` at the top, and then just assign a different value to it.